var/spack/repos/builtin/packages/py-pennylane-lightning/package.py

```diff
@@ -7,20 +7,123 @@
 from spack.package import *
 
 
-class PyPennylaneLightning(PythonPackage):
-    """PennyLane-Lightning plugin"""
+class PyPennylaneLightning(CMakePackage, PythonExtension):
+    """The PennyLane-Lightning plugin provides a fast state-vector simulator written in C++."""
 
-    homepage = "https://github.com/PennyLaneAI/pennylane-lightning"
-    url = "https://github.com/PennyLaneAI/pennylane-lightning/archive/refs/tags/v0.28.0.tar.gz"
-    # using github for now, because pypi tarball is missing the CMakeLists.txt file
-    # pypi = "PennyLane-Lightning/PennyLane-Lightning-0.28.0.tar.gz"
+    homepage = "https://docs.pennylane.ai/projects/lightning"
+    git = "https://github.com/PennyLaneAI/pennylane-lightning.git"
+    url = "https://github.com/PennyLaneAI/pennylane-lightning/archive/refs/tags/v0.28.2.tar.gz"
+    tag = "v0.28.2"
 
-    version("0.28.0", sha256="f5849c2affb5fb57aca20feb40ca829d171b07db2304fde0a37c2332c5b09e18")
+    maintainers = ["mlxd", "AmintorDusko"]
 
-    depends_on("python@3.8:", type=("build", "run"))
-    depends_on("py-setuptools", type="build")
-    depends_on("cmake@3.16:", type="build")
+    version("develop", branch="master")
+    version("0.28.2", sha256="c9b3afed0585681ccaf4df09fb12f2b7f09a8a1ba97a9b979139fe4a24509f31")
 
-    depends_on("py-ninja", type=("build", "run"))
-    depends_on("py-numpy", type=("build", "run"))
-    # depends_on("py-pennylane@0.19:", type=("build", "run"))  # circular dependency
+    patch(
+        "v0.28-spack_support.patch",
+        when="@0.28.2",
+        sha256="430078f3da285a10bd2d3a14bbdc41c7e8825be12ba713b6ca5dfca3fcc23d88",
+    )
+
+    variant("blas", default=True, description="Build with BLAS support")
+    variant(
+        "dispatcher",
+        default=True,
+        description="Build with AVX2/AVX512 gate automatic dispatching support",
+    )
+    variant("kokkos", default=True, description="Build with Kokkos support")
+    variant("openmp", default=True, description="Build with OpenMP support")
+    variant("python", default=True, description="Build with Python support")
+
+    variant("native", default=False, description="Build natively for given hardware")
+    variant("verbose", default=False, description="Build with full verbosity")
+
+    variant("cpptests", default=False, description="Build CPP tests")
+    variant("cppbenchmarks", default=False, description="Build CPP benchmark examples")
+
+    variant(
+        "build_type",
+        default="Release",
+        description="CMake build type",
+        values=("Debug", "Release", "RelWithDebInfo", "MinSizeRel"),
+    )
+
+    extends("python", when="+python")
+
+    # hard dependencies
+    depends_on("cmake@3.21.0:3.24.0", type="build")  # 3.21-3.24
```

Comment:
Recent issues we encountered with 3.25.0 and our Lightning suite caused us to hard-pin to 3.24 across our build ecosystem.

As 3.25.2 is now released, I do not see the issue. I can forbid the specific version here, rather than provide the upper bound.